src/Symfony/Component/Console/Application.php

```diff
@@ -941,6 +949,17 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
             // ignore invalid options/arguments for now, to allow the event listeners to customize the InputDefinition
         }
 
+        if ($this->signalRegistry) {
+            $event = new ConsoleSignalEvent($command, $input, $output);
+            $onStopHandler = function () use ($event) {
+                $this->dispatcher->dispatch($event, ConsoleEvents::SIGNAL);
+                /** @TODO what exit code to return? */
+                exit;
+            };
+            $this->signalRegistry->register(SIGINT, $onStopHandler);
```

Comment:
@lyrixx, how it would be convenient to set the list of handling signals from the outside?

Reply:
Use a static property + method on the registry ? **or** expose the registry on the command!

```php
class MyCommand
{
    public __construct() { $this->getApplication()->getSignalResitry()->enable(SIGINT); }
}
```

Reply:
Maybe it would be better to set signals for the application? Then we don't need to know about SignalRegistry in Commands. And through `SignalResitry()->enable(SIGINT)` I couldn't define a callback, thus this handler won't work until I assign callbacks. If we're going to use this Registry as a container for signals in whole project, there might be cases when one component wants to handle SIGINT with their own callback and another SIGUSR1 with another callback?